internal/cluster/common.go

```diff
@@ -190,8 +190,11 @@ func UpdateCluster(ctx context.Context, log logrus.FieldLogger, db *gorm.DB, eve
 	}
 
 	cluster, err := common.GetClusterFromDB(db, clusterId, common.UseEagerLoading)
-	if cluster != nil {
-		notifyEventStream(ctx, eventStream, &cluster.Cluster, log)
+	if cluster != nil && notificationStream != nil {
```

Comment:
I don't think you need the check for `notificationStream != nil` here, right?

Reply:
I've left it because we receive it as argument in the function, would you say we should let it blow up if nil is passed?

Reply:
Nevermind, just writing this I see there is no difference with the rest, why should we let it "fail" silently in this case? I'll update this line